zad1.js

```diff
@@ -0,0 +1,36 @@
+const superGra = fetch("https://raider.io/api/v1/characters/profile?region=eu&realm=burning-legion&name=Nunijo")
+const statusCheck = document.getElementById("status")
+
+superGra.then((response) => {
+    if (response.ok){
+        statusCheck.textContent = "status Promise : Fulfilled"    
+    }else{
+        statusCheck.textContent = "coś się popsuło i nie było mnie słychać"
+    }
+    return response.json()
+}, (error)=> {
+    console.log(response.ok)
+    statusCheck.textContent = "coś się popsuło i nie było mnie słychać"
+}).then((body)=>{
+    console.log(body)
+    const superMonke = body
+
+    for (let key in superMonke) {
+       const objKey = document.createElement("p")
+       objKey.id = `${key}`
+       objKey.textContent = `${key}: ${superMonke[key]}`
+       document.body.appendChild(objKey)
```

Comment:
Bardzo sprytny sposób na wypisanie tych rzeczy :). Co prawda nie zadziała gdy w payloadzie są zagnieżdżone obiekty - wtedy `${superMonke[key]}` będzie równy `[object Object]` bo taki jest wynik metody `toString()` na obiekcie.

Możesz albo:

* `console.log`nąć sobie to i zobaczyć pola które mogą być interesujące typu nie wiem, nazwa postaci, level, faction i je wypisać ręcznie korzystając z `body.<jakaśNazwaTamWłaściwości>`.
* Skorzystać z tego, że możesz ładnie zamienić obiekt JSONowy korzystając z metody `JSON.stringify` i wynik tego wywołania funkcji wrzucić jako `textContent`. JSON.stringify może "ładnie" sformatować JSONowy obiekt podając mu trzy parametry:

`JSON.stringify(obiekt, null, 2)`

Drugi parametr jest bezużyteczny dla wypisywania ładnie więc podajesz `null`, trzeci parametr mówi ile spacji ma użyć JS do wcięć - wcięcia zrobi automatycznie. Drugi i trzeci parametr są opcjonalne - bez nich wynikowy string z JSONem nie ma spacji i nowych linii - trudno się czyta, ale to poprawny JSON.

Reply:
czyli w tym przypadku to podejscie jest git, ale jakbym miał obiekt to cała pętla mi się wyjebie i ręcznie musze wybrać sobie tak jak mówisz np. z console log i oddzielnie po dodawać ?

Reply:
Tak